LuaMenu/widgets/api_user_handler.lua

```diff
@@ -1,993 +1,993 @@
+					clanDownloadBegun[clanName] = true
+				end
+				return clanFile, true
+			else
+				return false
+			end
+		end
+		return clanFile
+	end
+end
+
+--------------------------------------------------------------------------------
+--------------------------------------------------------------------------------
+-- Utilities that reference controls
+
+local function GetUserCountryImage(userName, userControl)
+	local userInfo = userControl.lobby:GetUser(userName) or {}
+	local userBattleInfo = userControl.lobby:GetUserBattleStatus(userName) or {}
+	if userInfo.country then
+		return CountryShortnameToFlag(userInfo.country)
+	end
+	if not userBattleInfo.aiLib then
+		return IMAGE_FLAG_UNKNOWN
+	end
+end
+
+local function GetUserSyncStatus(userName, userControl)
+	local userBattleInfo = userControl.lobby:GetUserBattleStatus(userName) or {}
+	if userBattleInfo.aiLib then
+		return
+	end
+	if userBattleInfo.sync == 1 then
+		return IMAGE_READY
+	else
+		return IMAGE_UNREADY
+	end
+end
+
+local function GetUserClanImage(userName, userControl)
+	local userInfo = userControl.lobby:GetUser(userName) or {}
+	local file, needDownload = GetClanImage(userInfo.clan)
+	return file, needDownload
+end
+
+local function GetUserComboBoxOptions(userName, isInBattle, userControl)
+	local userInfo = userControl.lobby:GetUser(userName) or {}
+	local userBattleInfo = userControl.lobby:GetUserBattleStatus(userName) or {}
+	local myUserName = userControl.lobby:GetMyUserName()
+	local comboOptions = {}
+
+	local myPartyID = userControl.lobby:GetMyPartyID()
+	local userPartyID = userControl.lobby:GetUserPartyID(userName)
+
+	local Configuration = WG.Chobby.Configuration
+
+	if (not userBattleInfo.aiLib) and userName ~= myUserName then
+		comboOptions[#comboOptions + 1] = "Message"
+
+		if (not isInBattle) and userInfo.battleID then
+			local battle = lobby:GetBattle(userInfo.battleID)
+			if battle and Configuration:IsValidEngineVersion(battle.engineVersion) then
+				if not Configuration.showMatchMakerBattles and battle.isMatchMaker then
+					comboOptions[#comboOptions + 1] = "Watch Battle"
+				else
+					comboOptions[#comboOptions + 1] = "Join Battle"
+				end
+			end
+		end
+
+		if (not myPartyID) or myPartyID ~= userPartyID then
+			-- Do not show any party options for people already in my party.
+			if (not myPartyID) and userPartyID then
+				-- Join others party if they have one and I don't.
+				comboOptions[#comboOptions + 1] = "Join Party"
+			else
+				-- Invite user to make a party or join mine. Note that the
+				-- user might be in a party which is not visible to me. In
+				-- this case the command might be the same as join party.
+				comboOptions[#comboOptions + 1] = "Invite to Party"
+			end
+		end
+
+		if Configuration.canAuthenticateWithSteam and userControl.steamInvite and userInfo.steamID then
+			comboOptions[#comboOptions + 1] = "Invite to Campaign"
+		end
+
+		if userInfo.accountID and Configuration.gameConfig.link_userPage ~= nil then
```

Comment:
Why are there so many changes that add "~= nil" in the checks for the existence of functions? They are ugly.

Reply:
I strongly prefer to be explicit in these checks, as otherwise it can mislead readers that they might be dealing with a boolean